packages/hardhat-ethers/test/index.ts

```diff
@@ -387,6 +387,41 @@ describe("Ethers plugin", function () {
             );
           });
 
+          it("should fail to create a contract factory when incorrectly linking a library with an ethers.Contract", async function () {
+            const libraryFactory = await this.env.ethers.getContractFactory(
+              "TestLibrary"
+            );
+            const library = await libraryFactory.deploy();
+
+            try {
+              const contractFactory = await this.env.ethers.getContractFactory(
+                "TestContractLib",
+                { libraries: { TestLibrary: library as any } }
+              );
+            } catch (reason) {
+              assert.instanceOf(
+                reason,
+                NomicLabsHardhatPluginError,
+                "getContractFactory should fail with a hardhat plugin error"
+              );
+              assert.isTrue(
+                reason.message.includes(
+                  "invalid address",
+                  "getContractFactory should report the invalid address as the cause"
+                )
+              );
+              assert.isTrue(
```

Comment:
I don't get this

Reply:
I added this assert to make sure the resulting string doesn't contain a stringification of an enormous object. The error attempts to be friendly in common cases by printing the passed in argument.

Reply:
Can you add a comment explaining why it's there?